src/doc/trpl/ffi.md

```diff
@@ -530,3 +530,37 @@ The `extern` makes this function adhere to the C calling convention, as
 discussed above in "[Foreign Calling
 Conventions](ffi.html#foreign-calling-conventions)". The `no_mangle`
 attribute turns off Rust's name mangling, so that it is easier to link to.
+
+# FFI and panics
+
+It’s important to be mindful of `panic!`s when working with FFI. This code,
+when called from C, will `abort`:
+
+```rust
+#[no_mangle]
+pub extern fn oh_no() -> ! {
+    panic!("Oops!");
+}
+# fn main() {}
+```
+
+If you’re writing code that may panic, you should run it in an other thread,
```

Comment:
@bors r+ rollup

On Sat, Jun 20, 2015 at 11:19 AM, Steve Klabnik <notifications@github.com>
wrote:

> In src/doc/trpl/ffi.md
> <https://github.com/rust-lang/rust/pull/26455#discussion_r32886642>:
>
> > @@ -530,3 +530,37 @@ The `extern` makes this function adhere to the C calling convention, as
> >  discussed above in "[Foreign Calling
> >  Conventions](ffi.html#foreign-calling-conventions)". The `no_mangle`
> >  attribute turns off Rust's name mangling, so that it is easier to link to.
> > +
> > +# FFI and panics
> > +
> > +It’s important to be mindful of `panic!`s when working with FFI. This code,
> > +when called from C, will `abort`:
> > +
> > +```rust
> > +#[no_mangle]
> > +pub extern fn oh_no() -> ! {
> > +    panic!("Oops!");
> > +}
> > +# fn main() {}
> > +```
> > +
> > +If you’re writing code that may panic, you should run it in an other thread,
>
> fixed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/26455/files#r32886642>.
>